src/coreclr/jit/lower.cpp

```diff
@@ -7065,7 +7065,16 @@ void Lowering::ContainCheckBitCast(GenTree* node)
 void Lowering::LowerStoreIndirCommon(GenTreeStoreInd* ind)
 {
     assert(ind->TypeGet() != TYP_STRUCT);
-    TryCreateAddrMode(ind->Addr(), true, ind);
+
+#if defined(TARGET_ARM64)
+    // Verify containment safety before creating an LEA that must be contained.
+    //
+    const bool isContainable = IsSafeToContainMem(ind, ind->Addr());
+#else
+    const bool     isContainable         = true;
+#endif
+    TryCreateAddrMode(ind->Addr(), isContainable, ind);
```

Comment:
`TryCreateAddrMode` is called with `true` from other places too so I wonder if those need this check as well

Reply:
We should be doing the "right" safety checks later when we actually mark the node contained.

So the issue is whether the target can handle a non-contained LEA. arm64 can't handle a non-contained LEA but xarch/arm32 can.